### PR TITLE
v4.4.0, fixes #30

### LINKED ISSUE #30: Bad dependency version python-engineio >=1.2.1
Issue:
`python-socketio` has dependency `python-engineio >=1.2.1`, described [here](https://github.com/conda-forge/python-socketio-feedstock/blob/master/recipe/meta.yaml#L30). `conda-forge` has `python-engineio 3.0.0` and trying to initialise `socketio.Client` fails with error:

`AttributeError: module 'engineio' has no attribute 'Client'`.

To replicate:

```
conda create -n socketioimport -c conda-forge python=3.7 python-socketio
conda activate socketioimport
python -c "import socketio; socketio.Client()"
```

The issue is not present on `pkgs/main` channel, as `python-engineio` is version 3.5. This leads me to believe that the lowest compatible version isn't 1.2.1, but something above 3.0

I'm installing `python-socketio` as part of a `conda-build`. If anyone else sees this, you will have to make the `anaconda` channel a higher priority than `conda-forge`. Instructions can be found here: [Managing channels](https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-channels.html#managing-channels)

<br/>
Environment (<code>conda list</code>):
<details>

```
$ conda list
# packages in environment at /home/dtasev/apps/miniconda3/envs/socketioimport:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main
ca-certificates           2019.6.16            hecc5488_0    conda-forge
certifi                   2019.6.16                py37_1    conda-forge
libedit                   3.1.20181209         hc058e9b_0
libffi                    3.2.1                hd88cf55_4
libgcc-ng                 9.1.0                hdf63c60_0
libstdcxx-ng              9.1.0                hdf63c60_0
ncurses                   6.1                  he6710b0_1
openssl                   1.1.1c               h516909a_0    conda-forge
pip                       19.1.1                   py37_0
python                    3.7.4                h265db76_1
python-engineio           3.0.0                      py_0    conda-forge
python-socketio           4.3.0                      py_0    conda-forge
readline                  7.0                  h7b6447c_5
setuptools                41.0.1                   py37_0
six                       1.12.0                py37_1000    conda-forge
sqlite                    3.29.0               h7b6447c_0
tk                        8.6.8                hbc83047_0
wheel                     0.33.4                   py37_0
xz                        5.2.4                h14c3975_4
zlib                      1.2.11               h7b6447c_3
```
</details>

<br/>
Details about  <code>conda</code> and system ( <code>conda info</code> ):
<details>

```
$ conda info
    active environment : socketioimport
    active env location : /home/dtasev/apps/miniconda3/envs/socketioimport
            shell level : 2
       user config file : /home/dtasev/.condarc
 populated config files : /home/dtasev/.condarc
          conda version : 4.7.11
    conda-build version : 3.18.9
         python version : 3.7.3.final.0
       virtual packages : __cuda=9.1
       base environment : /home/dtasev/apps/miniconda3  (writable)
           channel URLs : https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /home/dtasev/apps/miniconda3/pkgs
                          /home/dtasev/.conda/pkgs
       envs directories : /home/dtasev/apps/miniconda3/envs
                          /home/dtasev/.conda/envs
               platform : linux-64
             user-agent : conda/4.7.11 requests/2.22.0 CPython/3.7.3 Linux/4.18.0-25-generic ubuntu/18.04.2 glibc/2.27
                UID:GID : 1000:1000
             netrc file : None
           offline mode : False

```
</details>

### PR DESCRIPTION
**Pull request by ReimarBauer**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by ReimarBauer** [comment]
Hi @conda-forge/staged-recipes need help, can you merge my PR, and make me a co-maintainer?

A new developer using vmware in our project recognized that we need a newer version, because of failing tests in our project. Also an issue here was not resolved.

**Comment by beckermr** [comment]
You should bump @conda-forge/python-socketio for this!

**Comment by synapticarbors** [comment]
Should the `build` section be switched over to `host`? Also using the latest canonical `script` employing `pip`:

https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L35

(`pip` would need to be added to `host`)

**Comment by chrisburr** [comment]
Yes it should

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,35 +1,32 @@
...
 
 build:
-  noarch: python
   number: 0
-  script: python setup.py install --single-version-externally-managed --record=record.txt
+  noarch: python
+  script: "{{ PYTHON }} -m pip install . --no-deps -vv"
 
 requirements:
   build:
```
Comment (synapticarbors) [inline recipe/meta.yaml]: `build` -> `host`

**Comment by ReimarBauer** [comment]
> You should bump @conda-forge/python-socketio for this!

still learning. Thx for helping.

**Comment by ocefpaf** [comment]
Thank you for fixing this old feedstock! Glad we have a new maintainer here!

**Pull request merged**